web/src/upload.js

```diff
@@ -302,6 +302,48 @@ export function setup_upload(config) {
     $drag_drop_container.on("dragover", (event) => event.preventDefault());
     $drag_drop_container.on("dragenter", (event) => event.preventDefault());
 
+    $("#main_div").on("dragover", (event) => event.preventDefault());
+    $("#main_div").on("dragenter", (event) => event.preventDefault());
+
+    $("#main_div").on("drop", (event) => {
+        event.preventDefault();
+
+        // Check for edit box which registered the event handler no longer exists.
+        if (!$drag_drop_container.closest("html").length) {
+            return;
+        }
+
+        const $drag_drop_edit_containers = $(".message_edit_form form");
+        for (const container of $drag_drop_edit_containers) {
+            // This means file was dropped on #compose, which will be handled by different function
+            if (container.contains(event.target)) {
+                return;
+            }
+        }
+        // If the main compose box and edit boxes are all closed, open the compose box in reply mode and upload file.
+        // If compose box and edit box(es) are both open, the upload goes to main compose box.
+        // If only edit message boxes are open, the upload goes to the edit box at the bottom.
+        const $last_drag_drop_edit_container = $drag_drop_edit_containers.last();
+        const files = event.originalEvent.dataTransfer.files;
+        if (config.mode === "compose") {
```

Comment:
I don't think this outer if/else is coherent; see the above about how we don't want to do this initialization more than once. I think it should be a simple if/else if/else block with 3 possibilities: (1) `compose_state.composing()` => compose, (2) `$last_drag_drop_edit_container.length > 0` => that edit element, (3) else open compose with a reply state... maybe with a 4th if/else for what happens if you try to drag something onto say "Recent conversations" view or other state with no message selected.

Reply:
Should we listen for drag-drop on the `#app-main` or if we want to exclude the sidebars than `.column-middle`?
What should be the behaviour if file is dropped on sidebars?

Reply:
Yeah, I think `.app .column-middle` is probably fine as a place to start -- we can get feedback from there.

Reply:
After playing with this, I moved it to just `.app-main`; seems nicer to upload on drags to sidebars than any other option for what happens if you do that.